### ISSUE TITLE
ContextRoot path does not account for multiple domains using same path - feature suggestion

### ISSUE THREAD

**Issue by blasterpal** [ehazlett/interlock#158]
### Problem

https://github.com/ehazlett/interlock/blob/master/docs/interlock_data.md#context-root

Based on my experimentation, using `ContextRoot` doesn't support multiple hosts nor domains for same
path as stated. It would be nice to have context and domain/host.

The current code (v1.1.1) replaces `Domain` with the `ContextRoot`:  https://github.com/ehazlett/interlock/blob/b2b86fb042520535ba5e72a09f959c97cb57e733/ext/lb/haproxy/generate.go#L48 . Which means using the `Domain` or `HostName` in templates is not possible.
### Example

**(Note my current deployment is using the host to differentiate environments, but we could use part of domain)**

```
curl http://myapp-qa.company.co/api/v1/resource
curl http://myapp-acceptance.company.co/api/v1/resource
```
### Solution ideas
- Not mutating `Domain`
- Adding fields `interlock.context_domain` and/or `interlock.context_host`(this would leave current code path intact)

Example template use: 

```
 {{ range $host := .Hosts }}
 ...
# use both rules even on ContextRoot 
acl ctx_{{ $host.Name}} path_beg -i / {{ $host.ContextRoot.Path }}
acl is_{{ $host.Name }} hdr_beg(host) {{ $host.Domain }}

backend ctx_{{ $host.Name}} if is_{{ $host.Name }} ctx_{{ $host.Name}}
...
```
### TL;DR
1. Am I approaching this problem correctly with Interlock + HAProxy?
2. Should I have one HAProxy per environment (less ideal)?
3. Which solution is ideal? (I might take a stab at a PR)

Thanks,
Hank

**Comment by ehazlett** [comment]
When I first did the feature the requirements given were extremely vague. I think not mutating the domain is the best thing. Thanks for the feedback!

**Comment by ehazlett** [comment]
This should be fixed in #216.  Can you try `ehazlett/interlock:dev` to see if that fixes?  Thanks!

**Comment by ehazlett** [comment]
We are tracking most of this discussion in #206.  Closing this one.  Thanks!
